### PR TITLE
Fix second argument of reorder API to be more clear

### LINKED ISSUE #202: Fix reorder documentation for second argument
Using reorder() with what seems to be valid inputs makes the thread panic. 

MWE:
```rust
let val = [1., 2., 3., 4., 5., 6., 7., 8., 9., 10., 11., 12.];
let arr = Array::new(&val, Dim4::new(&[2, 3, 1, 2]));
let re_arr = reorder(&arr, Dim4::new(&[3, 2, 1, 2]));
```

Results in:

```
thread 'main' panicked at 'Error message: Size is incorrect
Last error: In function af_err af_reorder(af_array *, const af_array, const af::dim4 &)
In file src/api/c/reorder.cpp:84
Invalid dimension for argument 5
Expected: rdims[i] == allDims[rdims[i]]...
```

### PR DESCRIPTION
**Pull request by 9prady9**
Fixes #202 

@srenevey How about the updated API ? Also, do take a look at how new API handles it's second argument.

### CODE REVIEW

**Comment by 9prady9** [comment]
I will fix the point 1.

Regarding 2, I am still contemplating on what could be right way to pass
these new axes. If possible, I want to make sure I don't need to change the
API if max number of dimensions change in ArrayFire. I had a simple u64
argument earlier which was later removed. I will address this later today.

**Comment by 9prady9** [comment]
@srenevey I think it is more clear now. please have a look. 

Changing only one axis doesn't make sense, so I am debating if I take vec4<u64> altogether or what I have now. At least two axes are to be swapped, hence we have two mandatory u64 arguments and then an optional vec of u64's. This way, even if we change max dims support from ArrayFire, we can avoid any change in rust wrapper API.

File: src/data/mod.rs [hunk 1]
```diff
@@ -468,46 +469,112 @@ where
...
+/// Reorder the array in specified order
+///
+/// The default order of axes in ArrayFire is axis with smallest distance
+/// between adjacent elements towards an axis with highest distance between
+/// adjacent elements.
+///
+///# Parameters
+///
+/// - `input` is the input Array
+/// - `next_axes` is the new axeses order for output
```
Comment (srenevey) [inline src/data/mod.rs]: Typo in axeses. Should be axes

File: src/data/mod.rs [hunk 2]
```diff
@@ -468,46 +469,112 @@ where
...
+///
+///# Return Values
+///
+/// Array with data reordered as per the new axes order
+pub fn reorder_v2<T>(
```
Comment (srenevey) [inline src/data/mod.rs]: I would suggest `reorder<T>(input: &Array<T>, new_axes: Vec<u64>)` plain and simple. That way if new dimensions are added to ArrayFire you won't have to update this function. The signature that you propose could be added as a swap function such as `swap_axes<T>(input: &Array<T>, new_axis0: u64, new_axis1: u64)` in a future release for instance.

Comment (9prady9) [reply src/data/mod.rs]: `reorder` always requires at least two arguments - a minimum of two axes have to be provided to successfully swap data between two axes. Passing a single argument is invalid input and will only result in exception from ArrayFire upstream. Therefore, by enforcing two mandatory arguments for `reorder_v2` I am avoiding an unnecessary round trip to FFI land from rust, which also accurately reflects the functionality of reorder. Hence, the optional list of arguments (vec) is wrapped in `std::opt::Option`. For the prior mentioned reasons, I chose two mandatory u64 arguments followed by an optional vector which will cover all cases without any future modifications. We don't need another function like swap_axes at all.

**Pull request merged**